metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/database/SraXmlRetrieverThroughDatabase.java

```diff
@@ -47,6 +54,36 @@ public String getXml(String accession) {
         }
     }
 
+    @Override
+    public Map<String, String> getXmls(List<String> accessionList) {
+        Map<String, List> paramMap = new HashMap<>();
+        String sampleListQuery = enaObjectQuery;
+        String dynamicQuery = "SAMPLE_ID IN (:accession0) ";
+        List<List<String>> accessionSubList = Lists.partition(accessionList, MAX_QUERY_LIST);
+
+        // Oracle limitation: IN query can not have more than 1000 items in single list
+        for (int index = 0; index < (accessionList.size() / MAX_QUERY_LIST) + 1; index++) {
```

Comment:
we don't need this complex operation as Dylan suggested u can get all the sample XML from analysis_samples table using analysis accession

Reply:
I didn't find sample xml in `analysis_sample table`.
I can get sample accession for each analysis accession. From this i can get sample xml from `sample` table. Please let me know if 2 step DB access is fine.

Reply:
you can use join

Reply:
It does work with query "SELECT ANALYSIS_SAMPLE.SAMPLE_ID, SAMPLE.SAMPLE_XML FROM ANALYSIS_SAMPLE INNER JOIN SAMPLE ON ANALYSIS_SAMPLE.SAMPLE_ID = SAMPLE.SAMPLE_ID WHERE analysis_sample.analysis_id = 'xxx'";
The issue is when we pass `analysis id for getting samples`, code needs to be changed in ObjectImporter.importSamples() and further. This code becomes specific to EGA case. It wont work with API.
Workaround would be to know instantiated SRAObjectImporter object is API/DB and some tweaks. Code(importAnalysis) and methods becomes specific to call. Not sure this is ok.